### PR TITLE
Fix(test): log is not shown completely when node crashed in pytest

### LINKED ISSUE #2761: Pytests do not fail properly
When a pytest is expected to fail due to node crash, I observe the following:
```
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 421, in _make_request
    six.raise_from(e, None)
  File "<string>", line 3, in raise_from
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 416, in _make_request
    httplib_response = conn.getresponse()
  File "/usr/local/Cellar/python/3.7.4_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/http/client.py", line 1336, in getresponse
    response.begin()
  File "/usr/local/Cellar/python/3.7.4_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/http/client.py", line 306, in begin
    version, status, reason = self._read_status()
  File "/usr/local/Cellar/python/3.7.4_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/http/client.py", line 267, in _read_status
    line = str(self.fp.readline(_MAXLINE + 1), "iso-8859-1")
  File "/usr/local/Cellar/python/3.7.4_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/socket.py", line 589, in readinto
    return self._sock.recv_into(b)
socket.timeout: timed out

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/requests/adapters.py", line 449, in send
    timeout=timeout
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 720, in urlopen
    method, url, error=e, _pool=self, _stacktrace=sys.exc_info()[2]
  File "/usr/local/lib/python3.7/site-packages/urllib3/util/retry.py", line 400, in increment
    raise six.reraise(type(error), error, _stacktrace)
  File "/usr/local/lib/python3.7/site-packages/urllib3/packages/six.py", line 735, in reraise
    raise value
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 672, in urlopen
    chunked=chunked,
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 423, in _make_request
    self._raise_timeout(err=e, url=url, timeout_value=read_timeout)
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 331, in _raise_timeout
    self, url, "Read timed out. (read timeout=%s)" % timeout_value
urllib3.exceptions.ReadTimeoutError: HTTPConnectionPool(host='127.0.0.1', port=3041): Read timed out. (read timeout=5)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "tests/sanity/state_sync4.py", line 74, in <module>
    status1 = nodes[1].get_status()
  File "lib/cluster.py", line 124, in get_status
    r = requests.get("http://%s:%s/status" % self.rpc_addr(), timeout=5)
  File "/usr/local/lib/python3.7/site-packages/requests/api.py", line 75, in get
    return request('get', url, params=params, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/api.py", line 60, in request
    return session.request(method=method, url=url, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/sessions.py", line 533, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/sessions.py", line 646, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/adapters.py", line 529, in send
    raise ReadTimeout(e, request=request)
requests.exceptions.ReadTimeout: HTTPConnectionPool(host='127.0.0.1', port=3041): Read timed out. (read timeout=5)
```
and there is no indication of any error in the logs as well. It is not clear to me what happened and makes it very difficult to debug.

### PR DESCRIPTION
**Pull request by ailisp**
The root cause is once the node is panicked and exit, it takes a while for it to flush log to stdout and stderr (the pid still stays for one or two second before write finish, and even after os.kill it's there, until it write log finishes the node.pid disappear. So fix is in cleanup, check pid to ensure it's indeed finish. Fixes #2761 

Test Plan
----------
a short repo: add a if head.height>10 { panic!("aaa"); }, and run block_production.py. Before this PR log is truncated in test0_finished/stderr, but after this fix it's complete

### CODE REVIEW

File: pytest/lib/cluster.py [hunk 1]
```diff
@@ -324,6 +324,11 @@ def start(self, boot_key, boot_node_addr):
     def kill(self):
         if self.pid.value != 0:
             os.kill(self.pid.value, signal.SIGKILL)
+            while True:
+                try:
+                    os.kill(pid, 0)
+                except OSError:
```
Comment (bowenwang1996) [inline pytest/lib/cluster.py:330]: Is it possible for it to return any other error?

Comment (ailisp) [reply pytest/lib/cluster.py:330]: No, kill doesn't raise error if the pid exists, and raise OSError if pid doesn't exist

**Comment by bowenwang1996** [comment]
@ailisp state migration test is failing

**Comment by ailisp** [comment]
@bowenwang1996 some change for typo was forgot to commit, fortunately state_migration.py found it

**Pull request merged**